src/Courses/TopicSettings/ProblemSettings.tsx

```diff
@@ -143,13 +155,117 @@ export const ProblemSettings: React.FC<ProblemSettingsProps> = ({selected, setSe
         }
     };
 
+    const controls = useAnimation();
+    const [flipped, cycleFlipped] = useCycle(-1, 1);
+
     const rendererPreview = () => {
         return (
-            <ProblemIframe 
-                problem={new ProblemObject({  })} 
-                setProblemStudentGrade={() => { }} 
-                workbookId={0} 
-                readonly={false} />
+            <ExpansionPanel>
+                <ExpansionPanelSummary
+                    expandIcon={<ExpandMore />}
+                    aria-label="Expand"
+                    aria-controls="additional-actions3-content"
+                    id="additional-actions3-header"
+                >
+                    Problem Preview Pane
+                </ExpansionPanelSummary>
+                <ExpansionPanelDetails>
+                    <Grid container xs={12}>
+                        <Grid item xs={10}>
+                            <TextField
+                                fullWidth
+                                aria-label='Problem Path'
+                                variant='outlined'
+                                onClick={(event: any) => event.stopPropagation()}
+                                onFocus={(event: any) => event.stopPropagation()}
+                                onChange={(event: any)=>{
+                                    const val = event.target.value;
+                                    setPreviewSettings(settings => ({...settings, path: val}));
+                                }}
+                                onBlur={()=>{
+                                    // Validate first
+                                    // const regex = /^(Library|Contrib|webwork-open-problem-library|private\/our|private\/templates|private\/rederly).*\.pg$/;
+                                    // if (regex.test(previewSettings.path)) {
+                                        forceUpdate();
+                                    // }
+                                }}
+                                label='Problem Path'
+                                value={previewSettings.path}
+                                inputProps={{
+                                    pattern: /^(Library|Contrib|webwork-open-problem-library|private\/our|private\/templates|private\/rederly).*\.pg$/
+                                }}
+                            />
+                        </Grid>
+                        <Grid item xs={2}>
+                            {!_.isEmpty(pathArray?.additionalProblemPaths) && pathArray?.additionalProblemPaths?.[0].path}
```

Comment:
Do you technically need the array length check, if it's nillable anyway and your only using the first element you can always just null chain it